File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/Application/Omega/Feature.cs
using System;
using Psi.Data.Models.Domain;

namespace Psi.Data.Models.ClientConfigurationModels.Application.Omega
{
    public class Feature : SettingsBaseHelper
    {
        private AccountNumberAssociation _accountNumberAssociation;

        public Feature(ISettingsBase settingsBase) : base(settingsBase)
        {
        }

        [SettingKey("Omega.Features.RemoveBillSubscribersEnabled")]
        public bool RemoveBillSubscribersEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Omega.Features.XAppConfigEnabled")]
        public bool XAppConfigEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Omega.Features.StringResourcesEnabled")]
        public bool StringResourcesEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Omega.Features.ApplicationConfigEnabled")]
        public bool ApplicationConfigEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Omega.Features.AddNewUserEnabled")]
        public bool AddNewUserEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Omega.Features.EditUserPermissionsEnabled")]
        public bool EditUserPermissionsEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Omega.Features.VersionManagementEnabled")]
        public bool VersionManagementEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Omega.Features.QRCodeGeneratorEnabled")]
        public bool QRCodeGeneratorEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Omega.Features.LayeredSecurityEnabled")]
        public bool LayeredSecurityEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Omega.Features.FeaturesManagerEnabled")]
        public bool FeaturesManagerEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Omega.Features.LicenseManagerEnabled")]
        public bool LicenseManagerEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Omega.Features.CreditUnionContactInfoEnabled")]
        public bool CreditUnionContactInfoEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Omega.Features.OnlineBankingApiEnabled")]
        public bool OnlineBankingApiEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Omega.LayeredSecurity.MaximumRequiredMethods")]
        public int LayeredSecurityMaximumRequiredMethods
        {
            get { return GetIntValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Omega.LayeredSecurity.ExcludePasswordRule")]
        public bool LayeredSecurityExcludePasswordRule
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Omega.Features.ThirdPartySystemMessages.Enabled")]
        public bool EnableThirdPartySystemMessages
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Omega.Features.QuickAccessManagementEnabled")]
        public bool QuickAccessManagementEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Omega.Features.NewMobileFeaturesEnabled")]
        public bool NewMobileFeaturesEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }


        [SettingKey("Omega.Features.DeleteOnlineBankingUsersEnabled")]
        public bool DeleteOnlineBankingUsersEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }


        [SettingKey("Omega.Features.DeleteOnlineBankingUsersMaskAccountNumbers")]
        public bool DeleteOnlineBankingUsersMaskAccountNumbers
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }


        [SettingKey("Omega.Features.DeleteOnlineBankingUsersInitialDaysAgoToList")]
        public int DeleteOnlineBankingUsersInitialDaysAgoToList
        {
            get { return GetIntValue(); }
            set { SetValue(value); }
        }


        [SettingKey("Omega.Features.DeleteOnlineBankingUsersAllowCsvExportPriorToDeletingUsers")]
        public bool DeleteOnlineBankingUsersAllowCsvExportPriorToDeletingUsers
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }


        [SettingKey("Omega.Features.ConfigurationComparisonEnabled")]
        public bool ConfigurationComparisonEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Omega.Features.AuditLogExportTool.Enabled")]
        public bool AuditLogExportToolEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Omega.Features.DeleteBillPayUsers.Enabled")]
        public bool DeleteBillPayUsersEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Omega.Features.AuditLogExportTool.RemoteLoggingVendor")]
        public RemoteLoggingVendorType AuditLogExportRemoteLoggingVendor
        {
            get
            {
                Enum.TryParse(GetValue(), true, out RemoteLoggingVendorType result);
                return result;
            }
            set { SetValue(value); }
        }

        [SettingKey("Omega.Features.PermissionGroups.Enabled")]
        public bool PermissionGroupsEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Omega.Features.AtmBranchLocations.Enabled")]
        public bool AtmBranchLocationsEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("X.App.HBBOL.VerafinLoggingDir")]
        public string VerafinAuditLogExportToolOutputDirectory
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }

        [SettingKey("X.App.HBBOL.VerafinCustomerId")]
        public string VerafinCustomerId
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }

        [SettingKey("X.App.HBBOL.HostType")]
        public string HostType
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }

        public AccountNumberAssociation AccountNumberAssociation
        {
            get => _accountNumberAssociation ?? (_accountNumberAssociation = new AccountNumberAssociation(SettingsBase));
            set => _accountNumberAssociation = value;
        }

        //[SettingKey("Omega.Features.XAppConfigPermissionLevel")]
        //public PermissionLevel PermissionLevelXAppConfigEnabled
        //{
        //    get { return Enum.IsDefined(typeof(PermissionLevel), GetValue()) ? (PermissionLevel)Enum.Parse(typeof(PermissionLevel), GetValue()) : (PermissionLevel)Enum.Parse(typeof(PermissionLevel), "Basic"); }
        //    set { SetValue(value); }
        //}

        //[SettingKey("Omega.Features.StringResourcesPermissionLevel")]
        //public PermissionLevel StringResourcesPermissionLevel
        //{
        //    get { return Enum.IsDefined(typeof(PermissionLevel), GetValue()) ? (PermissionLevel)Enum.Parse(typeof(PermissionLevel), GetValue()) : (PermissionLevel)Enum.Parse(typeof(PermissionLevel), "Basic"); }
        //    set { SetValue(value); }
        //}

        //[SettingKey("Omega.Features.ApplicationConfigPermissionLevel")]
        //public PermissionLevel ApplicationConfigPermissionLevel
        //{
        //    get { return Enum.IsDefined(typeof(PermissionLevel), GetValue()) ? (PermissionLevel)Enum.Parse(typeof(PermissionLevel), GetValue()) : (PermissionLevel)Enum.Parse(typeof(PermissionLevel), "Basic"); }
        //    set { SetValue(value); }
        //}

        //[SettingKey("Omega.Features.AddNewUserPermissionLevel")]
        //public PermissionLevel AddNewUserPermissionLevel
        //{
        //    get { return Enum.IsDefined(typeof(PermissionLevel), GetValue()) ? (PermissionLevel)Enum.Parse(typeof(PermissionLevel), GetValue()) : (PermissionLevel)Enum.Parse(typeof(PermissionLevel), "Basic"); }
        //    set { SetValue(value); }
        //}

        //[SettingKey("Omega.Features.EditUserPermissionsPermissionLevel")]
        //public PermissionLevel EditUserPermissionsPermissionLevel
        //{
        //    get { return Enum.IsDefined(typeof(PermissionLevel), GetValue()) ? (PermissionLevel)Enum.Parse(typeof(PermissionLevel), GetValue()) : (PermissionLevel)Enum.Parse(typeof(PermissionLevel), "Basic"); }
        //    set { SetValue(value); }
        //}

        //[SettingKey("Omega.Features.LayeredSecurityPermissionLevel")]
        //public PermissionLevel LayeredSecurityPermissionLevel
        //{
        //    get { return Enum.IsDefined(typeof(PermissionLevel), GetValue()) ? (PermissionLevel)Enum.Parse(typeof(PermissionLevel), GetValue()) : (PermissionLevel)Enum.Parse(typeof(PermissionLevel), "Basic"); }
        //    set { SetValue(value); }
        //}
    }
}

File: workspace/legacy-analyzer/output/infrastructure/models/Omega/Feature.ts
// Generated imports
import { RemoteLoggingVendorType } from '../RemoteLoggingVendorType';
import { AccountNumberAssociation } from '../Application/Omega/AccountNumberAssociation';

export interface Feature {
    /** @settingKey Omega.Features.RemoveBillSubscribersEnabled */
    removeBillSubscribersEnabled: boolean;
    /** @settingKey Omega.Features.XAppConfigEnabled */
    xAppConfigEnabled: boolean;
    /** @settingKey Omega.Features.StringResourcesEnabled */
    stringResourcesEnabled: boolean;
    /** @settingKey Omega.Features.ApplicationConfigEnabled */
    applicationConfigEnabled: boolean;
    /** @settingKey Omega.Features.AddNewUserEnabled */
    addNewUserEnabled: boolean;
    /** @settingKey Omega.Features.EditUserPermissionsEnabled */
    editUserPermissionsEnabled: boolean;
    /** @settingKey Omega.Features.VersionManagementEnabled */
    versionManagementEnabled: boolean;
    /** @settingKey Omega.Features.QRCodeGeneratorEnabled */
    qRCodeGeneratorEnabled: boolean;
    /** @settingKey Omega.Features.LayeredSecurityEnabled */
    layeredSecurityEnabled: boolean;
    /** @settingKey Omega.Features.FeaturesManagerEnabled */
    featuresManagerEnabled: boolean;
    /** @settingKey Omega.Features.LicenseManagerEnabled */
    licenseManagerEnabled: boolean;
    /** @settingKey Omega.Features.CreditUnionContactInfoEnabled */
    creditUnionContactInfoEnabled: boolean;
    /** @settingKey Omega.Features.OnlineBankingApiEnabled */
    onlineBankingApiEnabled: boolean;
    /** @settingKey Omega.LayeredSecurity.MaximumRequiredMethods */
    layeredSecurityMaximumRequiredMethods: number;
    /** @settingKey Omega.LayeredSecurity.ExcludePasswordRule */
    layeredSecurityExcludePasswordRule: boolean;
    /** @settingKey Omega.Features.ThirdPartySystemMessages.Enabled */
    enableThirdPartySystemMessages: boolean;
    /** @settingKey Omega.Features.QuickAccessManagementEnabled */
    quickAccessManagementEnabled: boolean;
    /** @settingKey Omega.Features.NewMobileFeaturesEnabled */
    newMobileFeaturesEnabled: boolean;
    /** @settingKey Omega.Features.DeleteOnlineBankingUsersEnabled */
    deleteOnlineBankingUsersEnabled: boolean;
    /** @settingKey Omega.Features.DeleteOnlineBankingUsersMaskAccountNumbers */
    deleteOnlineBankingUsersMaskAccountNumbers: boolean;
    /** @settingKey Omega.Features.DeleteOnlineBankingUsersInitialDaysAgoToList */
    deleteOnlineBankingUsersInitialDaysAgoToList: number;
    /** @settingKey Omega.Features.DeleteOnlineBankingUsersAllowCsvExportPriorToDeletingUsers */
    deleteOnlineBankingUsersAllowCsvExportPriorToDeletingUsers: boolean;
    /** @settingKey Omega.Features.ConfigurationComparisonEnabled */
    configurationComparisonEnabled: boolean;
    /** @settingKey Omega.Features.AuditLogExportTool.Enabled */
    auditLogExportToolEnabled: boolean;
    /** @settingKey Omega.Features.DeleteBillPayUsers.Enabled */
    deleteBillPayUsersEnabled: boolean;
    /** @settingKey Omega.Features.AuditLogExportTool.RemoteLoggingVendor */
    remoteLoggingVendorType: RemoteLoggingVendorType;
    /** @settingKey Omega.Features.PermissionGroups.Enabled */
    permissionGroupsEnabled: boolean;
    /** @settingKey Omega.Features.AtmBranchLocations.Enabled */
    atmBranchLocationsEnabled: boolean;
    /** @settingKey X.App.HBBOL.VerafinLoggingDir */
    verafinAuditLogExportToolOutputDirectory: string;
    /** @settingKey X.App.HBBOL.VerafinCustomerId */
    verafinCustomerId: string;
    /** @settingKey X.App.HBBOL.HostType */
    hostType: string;
    accountNumberAssociation: AccountNumberAssociation;
}
